src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -67,12 +67,22 @@ public async Task<string> GetConnectionStringAsync()
         {
             if (_getConnectionStringFromSecretFunc != null)
             {
-                var secretProvider = _serviceProvider.GetRequiredService<ISecretProvider>();
-                return await _getConnectionStringFromSecretFunc(secretProvider);
+                return await GetConnectionStringFromSecretAsync();
             }
 
+            return GetConnectionStringFromConfiguration();
+        }
+
+        private string GetConnectionStringFromConfiguration()
+        {
             var configuration = _serviceProvider.GetRequiredService<IConfiguration>();
             return _getConnectionStringFromConfigurationFunc(configuration);
         }
+
+        private async Task<string> GetConnectionStringFromSecretAsync()
+        {
+            var secretProvider = _serviceProvider.GetRequiredService<ISecretProvider>();
+            return await _getConnectionStringFromSecretFunc(secretProvider);
```

Comment:
Should we null-check the returned `Task` before `await`ing? Could technically be `null`.

Reply:
How would you react to a null task according to you? Shouldn't we just bubble it or what are you thinking of?

Reply:
Yeah we could throw something, but something like a invalid arg, or post contract exception. Now it would be a null-ref without any info.